crates/oxc_linter/src/rules/jest/prefer_lowercase_title/mod.rs

```diff
@@ -0,0 +1,314 @@
+    /// title starting with an upper-case letter.
+    /// Example of **correct** code for the `{ "ignoreTopLevelDescribe": true }` option:
+    ///
+    /// ```js
+    /// /* eslint jest/prefer-lowercase-title: ["error", { "ignoreTopLevelDescribe": true }] */
+    /// describe('MyClass', () => {
+    ///     describe('#myMethod', () => {
+    ///         it('does things', () => {
+    ///             //
+    ///         });
+    ///     });
+    /// });
+    /// ```
+    ///
+    /// ### `lowercaseFirstCharacterOnly`
+    /// This option can be set to only validate that the first character of a test name is lowercased.
+    ///
+    /// Example of **correct** code for the `{ "lowercaseFirstCharacterOnly": true }` option:
+    ///
+    /// ```js
+    /// /* eslint vitest/prefer-lowercase-title: ["error", { "lowercaseFirstCharacterOnly": true }] */
+    /// describe('myClass', () => {
+    ///     describe('myMethod', () => {
+    ///         it('does things', () => {
+    ///             //
+    ///         });
+    ///     });
+    /// });
+    /// ```
+    ///
+    /// Example of **incorrect** code for the `{ "lowercaseFirstCharacterOnly": true }` option:
+    ///
+    /// ```js
+    /// /* eslint vitest/prefer-lowercase-title: ["error", { "lowercaseFirstCharacterOnly": true }] */
+    /// describe('MyClass', () => {
+    ///     describe('MyMethod', () => {
+    ///         it('does things', () => {
+    ///             //
+    ///         });
+    ///     });
+    /// });
+    /// ```
+    PreferLowercaseTitle,
+    style,
+    fix
+);
+
+impl Rule for PreferLowercaseTitle {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+        let ignore_top_level_describe = obj
+            .and_then(|config| config.get("ignoreTopLevelDescribe"))
+            .and_then(serde_json::Value::as_bool)
+            .unwrap_or(false);
+        let lowercase_first_character_only = obj
+            .and_then(|config| config.get("lowercaseFirstCharacterOnly"))
+            .and_then(serde_json::Value::as_bool)
+            .unwrap_or(true);
+        let ignore = obj
+            .and_then(|config| config.get("ignore"))
+            .and_then(serde_json::Value::as_array)
+            .map(|v| v.iter().filter_map(serde_json::Value::as_str).map(CompactStr::from).collect())
+            .unwrap_or_default();
+        let allowed_prefixes = obj
+            .and_then(|config| config.get("allowedPrefixes"))
+            .and_then(serde_json::Value::as_array)
+            .map(|v| v.iter().filter_map(serde_json::Value::as_str).map(CompactStr::from).collect())
+            .unwrap_or_default();
+
+        Self(Box::new(PreferLowercaseTitleConfig {
+            allowed_prefixes,
+            ignore,
+            ignore_top_level_describe,
+            lowercase_first_character_only,
+        }))
+    }
+
+    fn run_on_jest_node<'a, 'c>(
+        &self,
+        possible_vitest_node: &PossibleJestNode<'a, 'c>,
```

Comment:
lets keep this called `possible_vitest_node`

i think the idea here is to have jest rules, adapted for vitest not the other way round

Reply:
just to clarify, did you mean to say that we should keep it called `possible_jest_node`?

Reply:
I assume that's what you meant, so I renamed it to `possible_jest_node`.

Reply:
yeah that's right - thanks